src/core/infer_request.cc

```diff
@@ -249,18 +249,33 @@ InferenceRequest::CopyAsNull(const InferenceRequest& from)
     new_input->SetData(data);
   }
 
-
   // Second pass
   size_t max_byte_size = 0;
+  int64_t max_str_element_count = 0;
```

Comment:
Why not just use max_byte_size and max_input_name for both strings and non-strings. Then you just need a single alloc and don't need a lot of the other changes below? It will also be more performant and this code already has performance problems so should try not to make it slower.

Reply:
Addressed this comment in the next commit.